samples/Shared/UnitTestFramework/GenericAnalyzerTest.cs

```diff
@@ -0,0 +1,1355 @@
+        {
+            set
+            {
+                Assert.Empty(BatchFixedSources);
+                if (value != null)
+                {
+                    BatchFixedSources.Add(value);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the set of expected output files after a Fix All operation is applied. Files may be added to this list
+        /// using one of the <see cref="SourceFileList.Add"/> methods.
+        /// </summary>
+        /// <remarks>
+        /// <para>By default, Fix All operations are expected to produce the same result as incremental fix operations.
+        /// If this collection is not specified for the test, <see cref="FixedSources"/> provides the expected test
+        /// results for both incremental and Fix All scenarios.</para>
+        /// </remarks>
+        public SourceFileList BatchFixedSources { get; }
+
+        /// <summary>
+        /// Gets or sets the number of code fix iterations expected during code fix testing.
+        /// </summary>
+        /// <remarks>
+        /// <para>Code fixes are applied until one of the following conditions are met:</para>
+        ///
+        /// <list type="bullet">
+        /// <item><description>No diagnostics are reported in the input.</description></item>
+        /// <item><description>No code fixes are provided for the diagnostics reported in the input.</description></item>
+        /// <item><description>The code fix applied for the diagnostics does not produce a change in the source file(s).</description></item>
+        /// <item><description>The maximum number of allowed iterations is exceeded.</description></item>
+        /// </list>
+        ///
+        /// <para>If the number of iterations is positive, it represents an exact number of iterations: code fix tests
+        /// will fail if the code fix required more or fewer iterations to complete. If the number of iterations is
+        /// negative, the negation of the number of iterations is treated as an upper bound on the number of allowed
+        /// iterations: code fix tests will fail only if the code fix required more iterations to complete. If the
+        /// number of iterations is zero, the code fix test will validate that no code fixes are offered for the set of
+        /// diagnostics reported in the original input.</para>
+        ///
+        /// <para>When the number of iterations is not specified, the value is automatically selected according to the
+        /// current test configuration:</para>
+        ///
+        /// <list type="bullet">
+        /// <item><description>If the expected code fix output equals the input sources, the default value is treated as <c>0</c>.</description></item>
+        /// <item><description>Otherwise, the default value is treated as the negative of the number of fixable diagnostics appearing in the input source file(s).</description></item>
+        /// </list>
+        ///
+        /// <note>
```

Comment:
@dotnet/csharplangdesign This is an ideal situation where dotnet/csharplang#1767 would help